pipelines/stripe_analytics/stripe_analytics.py

```diff
@@ -0,0 +1,107 @@
+from datetime import datetime
+from enum import Enum
+from typing import Any, Generator, Optional, Dict
+
+import dlt
+import stripe
+from dlt.extract.source import DltResource
+from dlt.pipeline import Pipeline
+from dlt.common import pendulum
+
+from .metrics import calculate_mrr, churn_rate
+
+
+class Endpoints(Enum):
+    customers: str = "Customer"
+    subscriptions: str = "Subscription"
+    prices: str = "Price"
+    discounts: str = "Coupon"
+    events: str = "Event"
+
+
+def stripe_get_data(
+    resource: Endpoints, start_date: Optional[Any] = None, end_date: Optional[Any] = None, **kwargs : Any
+) -> Dict[Any, Any]:
+    if start_date:
+        if isinstance(start_date, str):
+            start_date = datetime.strptime(start_date, "%Y-%m-%dT%H:%M:%SZ")
+        if isinstance(start_date, datetime):
+            # convert to unix timestamp
+            start_date = int(start_date.timestamp())
+    if end_date:
+        # convert to unix timestamp
+        end_date = int(end_date.timestamp())
+
+    if resource == Endpoints.subscriptions:
+        kwargs.update({"status": "all"})
+
+    resource_dict = getattr(stripe, resource.value).list(
+        created={"gte": start_date, "lt": end_date}, **kwargs
+    )
+    return dict(resource_dict)
+
+
+@dlt.source
+def stripe_source(
+    stripe_secret_key: str = dlt.secrets.value,
+    limit: int = 100,
+    get_all_data: bool = False,
+    start_date: Optional[datetime] = None,
+    end_date: Optional[datetime] = None,
+) -> Generator[DltResource, Any, None]:
+    stripe.api_key = stripe_secret_key
+    stripe.api_version = "2022-11-15"
+
+    def get_resource(
+        endpoint: Endpoints,
+        created: Optional[Any] = dlt.sources.incremental("created", initial_value=-3600),
```

Comment:
IMO you should use `start_date` as initial value of incremental. then you will be consistent with other pipelines. we had a long discussion with @adrianbr on how we should deal with initial values in incremental load. and this is what we decided:
1. if the start_date is passed it is used as initial value for a first load (full load)
2. on subsequent loads we continue from the `last_value` and ignore start date
3. people can reset the resource/source by doing `write_dispositon="replace"` as described here https://dlthub.com/docs/general-usage/source#do-a-full-refresh